### ISSUE TITLE
yaml 1.1 parses single n or y character as boolean

### ISSUE THREAD

**Issue by stefanprobst** [netlify/netlify-cms#3893]
**Describe the bug**
I just ran into this weird issue that YAML v1.1 will apparently parse a single unquoted `y` or `n` character as `true` or `false`.

**To Reproduce**
```node
> yaml.parse('name: y', { version: '1.2' })
{ name: 'y' }
> yaml.parse('name: y', { version: '1.1' })
{ name: true }
```

**Expected behavior**
"n" and "y" character on a string field in a yaml file should be parsed as strings

**Applicable Versions:**
 - Netlify CMS version: 2.12.14

**Potential solutions**
Netlify CMS intentionally uses YAML v1.1 for parsing ([see here](https://github.com/netlify/netlify-cms/blob/374783509b506ab71a63fca1b2f630798101704a/packages/netlify-cms-core/src/formats/yaml.js#L25)). We could either use YAML 1.2, or save "n" and "y" with quotes on yaml string fields

**Comment by stefanprobst** [comment]
there are also differences in date parsing between 1.1 and 1.2
```node
> yaml.parse('name: 2020-06-12', { version: '1.2' })
{ name: '2020-06-12' }
> yaml.parse('name: 2020-06-12', { version: '1.1' })
{ name: 2020-06-12T00:00:00.000Z }
> yaml.parse('name: 12-06-2012', { version: '1.2' })
{ name: '12-06-2012' }
> yaml.parse('name: 12-06-2012', { version: '1.1' })
{ name: '12-06-2012' }
```

this means that on a `datetime` widget which sets `format: YYYY-MM-DD` this will currently be parsed as a Date object, while setting any other format will be parsed as a string

**Comment by testpad** [comment]
What happened to the original question for this issue - that of the unquoted Y or N?
The implication is you can't have a string widget that saves a single letter string when it's a Y or an N. Or rather, it saves just fine (without quotes), but you can't then reload the page because that field has been parsed as a boolean.

**Comment by sbutlin** [comment]
What happened to the original question for this issue - that of the unquoted Y or N?
The implication is you can't have a string widget that saves a single letter string when it's a Y or an N. Or rather, it saves just fine (without quotes), but you can't then reload the page because that field has been parsed as a boolean.

**Comment by erezrokah** [comment]
The original issue should have been fixed in #3923. See https://github.com/netlify/netlify-cms/pull/3923/files#diff-8eae933bb10932427533ce405fd808e9R64

If that is not the case please open a new issue.
